harness/determined/exec/prep_container.py

```diff
@@ -255,6 +267,9 @@ def do_proxy(sess: Session, allocation_id: str) -> None:
 
     if args.resources:
         det.ResourcesInfo._by_inspection()._to_file()
+        from determined.gpu import check_for_gpu_processes
+
+        check_for_gpu_processes()
```

Comment:
Can we separate the business logic (warning the user) from the helper logic (listing processes from nvidia-smi)?

Basically, let the gpu module have a `get_gpu_processes()` call, but emit the warning from here.